python/tvm/script/parser.py

```diff
@@ -14,21 +14,22 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-"""TVM Script Parser For TIR"""
-# pylint: disable=invalid-name, missing-docstring, inconsistent-return-statements, no-else-return
-# pylint: disable=unnecessary-comprehension, unused-argument
-# pylint: disable=relative-beyond-top-level
+"""TVM Script Parser For TIR
+
+We use [synr](https://synr.readthedocs.io) to get an AST that is stable over
+different python versions. Synr also provides an error handling context that we
+use for error reporting.
+"""
+# pylint: disable=invalid-name, inconsistent-return-statements, no-else-return
 import json
 import operator
 import inspect
-from typed_ast import ast3 as ast
+from synr import ast, Transformer, to_ast
```

Comment:
Out of curiosity, are we deprecating all usage of `typed_ast` in favour of `synr`? If so, I think we also want to replace and make this dependency official, here:

https://github.com/apache/incubator-tvm/blob/c8064b3ca6787961c90fe7c09c3ddc3beba3ece5/python/setup.py#L186

Plus remove `typed_ast` from the python packages script, maybe in a follow-up patch to support the transition period:

https://github.com/apache/incubator-tvm/blob/c8064b3ca6787961c90fe7c09c3ddc3beba3ece5/docker/install/ubuntu_install_python_package.sh#L24

In case you're doing it in a follow-up patch, can you add a TODO comment?

Reply:
Synr is listed in `incubator-tvm/docker/install/ubuntu_install_python_package.sh` (this branch wasn't up to date with main). We think we might have to update it often in the future, so we are installing a pinned version in the ci before we run tests (in `tests/scripts/task_ci_python_setup.sh`).

Reply:
I'm not sure we need a transition period for removing typed_ast. My impression is that tvmscript is not yet stabilized. Maybe @tqchen can chime in here?

Reply:
> Synr is listed in incubator-tvm/docker/install/ubuntu_install_python_package.sh (this branch wasn't up to date with main). We think we might have to update it often in the future, so we are installing a pinned version in the ci before we run tests (in tests/scripts/task_ci_python_setup.sh).

I guess my point is that if an user installs the tvm python module with setuptools or as a package, they won't get Synr as a dependency.

Reply:
Good point. I've added synr to `setup.py`.

Reply:
I think required, as `typed_ast` was.